packages/edit-post/src/components/layout/style.scss

```diff
@@ -68,11 +68,58 @@
 }
 
 .edit-post-layout__content {
-	position: relative;
 	display: flex;
-	min-height: 100%;
 	flex-direction: column;
 
+	// These rules are specific to mobile and small breakpoints.
+	min-height: 100%;
+	position: relative;
+
+	// We scroll the main editing canvas, the sidebar, and the block library separately to prevent scroll bleed.
+	// Because the navigation sidebar menu has "flyout" menus, we can't yet, scroll that independently, but as soon
+	// as we can, we should simplify these rules.
+	// In the mean time, if a user has a small screen and lots of plugin-added menu items in the navigation menu,
+	// they have to be able to scroll. To accommodate the flyout menus, we scroll the `body` element for this.
+	@include break-medium() {
+		// Because the body element scrolls the navigation sidebar, we have to use position fixed here.
+		// Otherwise you would scroll the editing canvas out of view when you scroll the sidebar.
+		position: fixed;
```

Comment:
There's an issue related to this change I think that cause the columns block to be very tight in the e2e tests. The left position of the editor is not correct I think.

Instead of duplicating all the `.folded`, `.auto-fold`... styles, can't we use the `.editor-left` mixin as I think it's meant to address exactly this issue.

Reply:
We probably could, yes, but the mixin sets the `left` property, whereas we have to set the `margin-left` property. 

It's complicated to explain why we need to use margin instead of the left property, but the reason is the same as why the sidebar uses it. The sidebar is fixed position using the same method as now the editing canvas, and by using margins we don't have to account for whether a scrollbar is present or not. 

For example the notices use `editor-right`, and for that reason they overlap scrollbars, which is one of the things that were fixed in #12301. You can see that the sidebar also uses margins for that reason, here: https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/sidebar/style.scss#L114

I suppose I can create a new mixin, `@editor-margin-left` that is similar to `@editor-left`, do you think that would be a good approach?

Reply:
I don't know, if it's specific to this component maybe not worth-it. But it would be good to detect why the `margin-left` is wrong in the e2e tests. 

We're probably missing a case here.

In the e2e test, I see the WP left menu as collapsed but the editor toolbar and the editor content left position is wrong, it's shown as if the left menu is not collapsed.

Reply:
Let me take a look again to see if the rules are right.

Reply:
Just tried to follow the instructions in the test all manually, and they fail for me too. However they fail the same for me in master and in this branch. It's only the e2e test that fails in this branch:

![test in this branch](https://user-images.githubusercontent.com/1204802/50284737-4607e100-045a-11e9-95df-cfedc9df8b53.gif)

I'm really confused why this test fails. 

I've tested thoroughly the left margin now, and I cannot get it to a state where it isn't right, with regards to the state of the left menu.

Reply:
> In the e2e test, I see the WP left menu as collapsed but the editor toolbar and the editor content left position is wrong, it's shown as if the left menu is not collapsed.

How did you get to this state? I can't reproduce it.

Reply:
The thing is I couldn't reproduce as well unless I run the e2e test in non-headless mode.
I think it has to do with:

 - First time you visit the admin (no persisted preference for the sidebar...)
 - Size of the window should be something around 960x700

Reply:
Awesome, thank you. Trying.

Reply:
Oh wow this is super interesting. At that breakpoint, the text "Second column paragraph" wraps into two lines:

![screenshot 2018-12-20 at 13 53 24](https://user-images.githubusercontent.com/1204802/50286137-bc0e4700-045e-11e9-9587-a19db42a764c.png)

That means that you _do_ need THREE presses of the up arrow to go to the 1st column. Whereas when there's enough space in the main editing canvas for the text to not wrap, you only have to press the up arrow TWICE.

Regardless of the test failure, or this patch I'm currently testing, this means the test currently only works on a very specific screen size.

Reply:
So, here's a GIF of me reproducing master at about that screen size:

![master](https://user-images.githubusercontent.com/1204802/50286803-d9441500-0460-11e9-93e5-685bbe0ecd0f.gif)

Here's me manually reproducing the same test in this branch:

![this branch](https://user-images.githubusercontent.com/1204802/50286815-e06b2300-0460-11e9-928b-c7e5e15925ab.gif)

The bit that is failing is that once the caret is at the end of the second column paragraph, you press "Up" arrow _three times_. One to go to the first line, another to select the _column_, and a third to go to the 1st column paragraph. 

This is the same in both branches when I test. 

But if the viewport is this small, then the words "Second column paragraph" wrap onto three lines:

![screenshot 2018-12-20 at 14 08 53](https://user-images.githubusercontent.com/1204802/50286873-0c86a400-0461-11e9-8c04-9b078b7fae1b.png)

If that were the case, then it would produce the expected result from our test failure, i.e. return "Second column paragraph", because the 2nd _column_ would be selected.